app.js

```diff
@@ -20,72 +19,69 @@ app.get('/', function (req, res) {
 });
 
 // creates an authentication token
-app.get('/auth-token.json', (req, res) => {
+app.get('/auth-token.json', async (req, res) => {
 
   console.log('creating new temp auth token');
 
-  //more information at https://docs.scanii.com/v2.1/resources.html
-  const options = {
-    url: 'https://api.scanii.com/v2.1/auth/tokens',
-    auth: {
-      'user': SCANII_CREDS.split(':')[0],
-      'pass': SCANII_CREDS.split(':')[1]
-    },
-    method: 'POST',
-    form: {
-      timeout: 300
-    }
-  };
-
-  request(options, (error, response, body) => {
-
-
-    if (response.statusCode === 201) {
-      let token = JSON.parse(body);
+  try {
+    //more information at https://docs.scanii.com/v2.1/resources.html
+    const credentials = Buffer.from(`${SCANII_CREDS.split(':')[0]}:${SCANII_CREDS.split(':')[1]}`).toString('base64');
+    
+    const response = await fetch('https://api.scanii.com/v2.1/auth/tokens', {
+      method: 'POST',
+      headers: {
+        'Authorization': `Basic ${credentials}`,
+        'Content-Type': 'application/x-www-form-urlencoded'
+      },
+      body: new URLSearchParams({
+        timeout: '300'
+      })
+    });
+
+    if (response.status === 201) {
+      const token = await response.json();
       console.log(`token created successfully with id:${token.id} and expiration: ${token.expiration_date}`);
       res.json(token);
-    }
-
-    else {
+    } else {
+      const body = await response.text();
       console.error(`error response from server while creating token`);
-      console.error(`http status: ${response.statusCode} message: ${body}`);
+      console.error(`http status: ${response.status} message: ${body}`);
       res.status(500).end();
     }
-  })
-
-
+  } catch (error) {
+    console.error('Error creating token:', error);
+    res.status(500).end();
+  }
 });
 
 // handles the final post
-app.post('/process', upload.single('file'), (req, res) => {
+app.post('/process', upload.single('file'), async (req, res) => {
   console.log('ensuring file has been properly processed by looking it up by the file id');
   const fileId = req.body.fileId || res.status(400).send('content does not appear to have been client-side processed');
   const uploadedFile = req.file || res.status(400).send('content does not appear to have been client-side processed');
 
+  try {
+    // now that we have the file id, we look it up in scanii to ensure no findings
+    // https://docs.scanii.com/v2.1/resources.html
+    const credentials = Buffer.from(`${SCANII_CREDS.split(':')[0]}:${SCANII_CREDS.split(':')[1]}`).toString('base64');
+    
+    const response = await fetch(`https://api.scanii.com/v2.1/files/${fileId}`, {
```

Comment:
change this url to https://api.scanii.com/v2.2/files/${fileId} - the api contract is the same so nothing else needs to change

Reply:
Upgrade the auth token to v2.2 as well please